plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -166,20 +174,35 @@ func PrioritizeNodes(pod *api.Pod, podLister algorithm.PodLister, priorityConfig
 		priorityFunc := priorityConfig.Function
 		prioritizedList, err := priorityFunc(pod, podLister, minionLister)
 		if err != nil {
-			return algorithm.HostPriorityList{}, err
+			return schedulerapi.HostPriorityList{}, err
 		}
 		for _, hostEntry := range prioritizedList {
 			combinedScores[hostEntry.Host] += hostEntry.Score * weight
 		}
 	}
+	if extender != nil && minionLister != nil {
+		nodes, err := minionLister.List()
+		if err != nil {
+			return schedulerapi.HostPriorityList{}, err
+		}
+
+		prioritizedList, err := extender.Prioritize(pod, &nodes)
```

Comment:
Regarding first comment: I think the APIs should be the same. So I would suggest passing all the pods to the extender.

Regarding second comment: Sorry, you're right, never mind my comment.